homeassistant/components/hassio/ingress.py

```diff
@@ -117,24 +117,38 @@ async def _handle_request(
     ) -> web.Response | web.StreamResponse:
         """Ingress route for request."""
         url = self._create_url(token, path)
-        data = await request.read()
         source_header = _init_header(request, token)
 
+        def _with_chunks(origin: web.Request | web.Response) -> bool:
+            return (
+                hdrs.CONTENT_LENGTH not in origin.headers
+                or int(origin.headers[hdrs.CONTENT_LENGTH]) > 4_194_000
+            )
+
+        async def _prepare_content():
+            if not _with_chunks(request):
+                data = await request.read()
+                yield data
+            else:
+                # Disable max size, we are chunking large requests anyway
+                request._client_max_size = None  # pylint: disable=protected-access
+                async for data in request.content.iter_any():
+                    yield data
+
         async with self._websession.request(
             request.method,
             url,
             headers=source_header,
             params=request.query,
             allow_redirects=False,
-            data=data,
+            chunked=_with_chunks(request),
+            data=_prepare_content(),
```

Comment:
Can't we just do this:

From https://docs.aiohttp.org/en/stable/client_quickstart.html#streaming-uploads:

> Because the content attribute is a StreamReader (provides async iterator protocol), you can chain get and post requests together:
>
> ```python
> resp = await session.get('http://python.org')
> await session.post('http://httpbin.org/post', data=resp.content)
> ```